Assets/MixedRealityToolkit/Utilities/Editor/PackageManifest/PackageManifestUpdater.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor
+{
+    /// <summary>
+    /// Class that enables MRTK to add scoped registries and/or packages to the Unity Package Mangager manifest
+    /// </summary>
+    internal static class PackageManifestUpdater
+    {
+        private static string PackageManifestRelativePath = Path.Combine("Packages", "manifest.json");
+
+        /// <summary>
+        /// Ensures the required settings exist in the package manager to allow for
+        /// installing MSBuild for Unity.
+        /// </summary>
+        internal static void EnsureMSBuildForUnity()
+        {
+            // Locate the full path to the package manifest.
+            DirectoryInfo projectRoot = new DirectoryInfo(Application.dataPath).Parent;
+            string manifestPath = Path.Combine(projectRoot.FullName, PackageManifestRelativePath);
+            
+            // Verify that the package manifest file exists.
+            if (!File.Exists(manifestPath))
+            {
+                Debug.LogError("Unable to locate the package manifest file");
+                return;
+            }
+
+            PackageManifest manifest = null;
+
+            // Read the package manifest into a list of strings (for easy finding of entries)
+            // and then deserialize.
+            List<string> manifestFileLines = new List<string>();
+            using (FileStream manifestStream = new FileStream(manifestPath, FileMode.Open, FileAccess.Read))
+            {
+                using (StreamReader reader = new StreamReader(manifestStream))
```

Comment:
Do we need the reader for this or could we use [`File.ReadAllLines`](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.readalllines?view=netframework-4.8)?

Reply:
the reader lets us read each line for later writing one at a time.